File: php/examples/zcsv_status.php
<?php
/**
 * Example of using ZippedCsv to check the status of a zipped csv file.
 * Usage:
 *  php examples/zcsv_status.php <path_to_zipped_csv_file>
 * App will print:
 * 
 * - metadata
 * - list of csv files in the zipped file
 * - for each csv file print number of rows and columns
 */


include __DIR__ . '/../src/mc/csv.php';
include __DIR__ . '/../src/mc/zippedcsv.php';

use \mc\ZippedCsv;

if ($argc < 2) {
    echo "Usage: php examples/zcsv_status.php <path_to_zipped_csv_file>" . PHP_EOL;
    exit(1);
}

if(!file_exists($argv[1])) {
    echo "File not found: {$argv[1]}" . PHP_EOL;
    exit(1);
}

try {
    $zcsv = new ZippedCsv($argv[1]);
    echo "Metadata:" . PHP_EOL;
    echo json_encode($zcsv->GetMetadata(), JSON_PRETTY_PRINT) . PHP_EOL;

    $csvFiles = $zcsv->GetTableNames();
    echo "CSV files in the zipped file:" . PHP_EOL;

    foreach ($csvFiles as $csvFile) {
        $csv = $zcsv->GetCsv($csvFile);
        echo "  - {$csvFile}, rows: " . $csv->TotalRows()
            . ", columns: " . $csv->TotalColumns() . PHP_EOL;
    }

    $zcsv->Close();
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . PHP_EOL;
    exit(1);
}

File: python/examples/zcsv_status.py
from __future__ import annotations

import json
import os
import sys

from zippedcsv import ZippedCsv


def main() -> int:
    if len(sys.argv) < 2:
        print("Usage: python examples/zcsv_status.py <path_to_zipped_csv_file>")
        return 1

    zip_file = sys.argv[1]
    if not os.path.exists(zip_file):
        print(f"File not found: {zip_file}")
        return 1

    try:
        zcsv = ZippedCsv(zip_file)
        print("Metadata:")
        print(json.dumps(zcsv.GetMetadata(), indent=2, ensure_ascii=False))

        csv_files = zcsv.GetTableNames()
        print("CSV files in the zipped file:")
        print(csv_files)

        for csv_file in csv_files:
            csv = zcsv.GetCsv(csv_file)
            print(f"CSV file: {csv_file}")
            print(f"Number of rows: {csv.TotalRows()}")
            print(f"Number of columns: {csv.TotalColumns()}")
        return 0
    except Exception as error:
        print(f"Error: {error}")
        return 1


if __name__ == "__main__":
    raise SystemExit(main())
